### PR TITLE
feat: remove ogen and client refactor

### PR DESCRIPTION
**Pull request by matoszz**
This PR encompasses a number of areas, but notably:
- full removal of the entoas / ogen libraries / plugins, their associated schema annotations, the openapi.json output and associated tasks / jobs
- re-implementation of OpenAPI specification generation via a light weight wrapper `Router` around Echo and the OpenAPI library, refactored routes, handlers, and collapse of many sever options and setups in favor of AddRoute off of echo groups (vs. the ForGroup statements on the individual routes) and the Router wrapper
- refactored `datumclient` which provides a dramatically simpler client instantiation, access to both the Graph and REST client interfaces, helper functions for better client cookie interactions, session interactions, token interactions
- creation of `pkg/models` - centralizing our Request and Response structs, examples, and validation

### CODE REVIEW

File: internal/httpserve/handlers/resendemail.go [hunk 1]
```diff
@@ -90,14 +82,29 @@ func (h *Handler) ResendEmail(ctx echo.Context) error {
...
-func validateResendRequest(req ResendRequest) error {
+func validateResendRequest(req models.ResendRequest) error {
 	if req.Email == "" {
 		return rout.NewMissingRequiredFieldError("email")
 	}
 
 	return nil
 }
+
+// BindResendEmail binds the resend email verification endpoint to the OpenAPI schema
```
Comment (golanglemonade) [inline internal/httpserve/handlers/resendemail.go:97]:
```suggestion
// BindResendEmailHandler binds the resend email verification endpoint to the OpenAPI schema
```

File: internal/httpserve/handlers/resetpassword.go [hunk 2]
```diff
@@ -181,3 +153,18 @@ func (u *User) setResetTokens(user *generated.User, reqToken string) error {
 	// be there
 	return ErrPassWordResetTokenInvalid
 }
+
+// BindResetPassword binds the reset password handler to the OpenAPI schema
```
Comment (golanglemonade) [inline internal/httpserve/handlers/resetpassword.go:157]:
```suggestion
// BindResetPasswordHandler binds the reset password handler to the OpenAPI schema
```

File: internal/httpserve/handlers/switch.go [hunk 3]
```diff
@@ -137,3 +126,18 @@ func switchClaims(u *generated.User, targetOrgMappingID string) *tokens.Claims {
 		OrgID:  targetOrgMappingID,
 	}
 }
+
+// BindResetPassword binds the reset password handler to the OpenAPI schema
```
Comment (golanglemonade) [inline internal/httpserve/handlers/switch.go:130]:
```suggestion
// BindSwitchHandler binds the reset password handler to the OpenAPI schema
```

File: internal/httpserve/route/events.go [hunk 4]
```diff
@@ -4,22 +4,28 @@ import (
...
+	route := echo.Route{
+		Name:   name,
+		Method: method,
+		Path:   path,
 		Handler: func(c echo.Context) error {
-			return h.EventPublisher(c)
+			return router.Handler.EventPublisher(c)
 		},
-	}.ForGroup(V1Version, authMW))
+	}.ForGroup(V1Version, mw)
```
Comment (golanglemonade) [inline internal/httpserve/route/events.go:22]: this now longer has the auth middleware, was that intentional?

File: internal/httpserve/route/forgotpass.go [hunk 5]
```diff
@@ -4,43 +4,27 @@ import (
...
-	}.ForGroup(V1Version, restrictedEndpointsMW))
-
-	if err != nil {
-		return err
-	}
-
-	// we need to handle the redirect from /.well-known 302 -> /v1/forgot-password with a 200
-	_, err = router.AddRoute(echo.Route{
-		Name:   "ForgotPasswordRedirect",
-		Method: http.MethodGet,
```
Comment (golanglemonade) [inline internal/httpserve/route/forgotpass.go:32]: Why are you dropping the redirect?

File: internal/httpserve/server/openapi.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+	"SwitchRequest":              &models.SwitchOrganizationRequest{},
+	"SwitchResponse":             &models.SwitchOrganizationReply{},
+	"VerifySubscriptionRequest":  &models.VerifySubscribeRequest{},
+	"VerifySubscriptionResponse": &models.VerifySubscribeReply{},
+	"InviteRequest":              &models.InviteRequest{},
+	"InviteResponse":             &models.InviteReply{},
+	"ErrorResponse":              &rout.StatusError{},
+}
+
+// SecurityScheme is an interface that represents a security scheme
```
Comment (golanglemonade) [inline internal/httpserve/server/openapi.go:177]: Should be a comment about `OAuth2`,  not security scheme

File: internal/httpserve/route/forgotpass.go (hunk 5 above)
Comment (matoszz) [reply internal/httpserve/route/forgotpass.go:32]: well, it doesn't actually work and i'm not certain why so I figured I would just remove it until I had time to debug the redirect middleware

**Review by golanglemonade — APPROVED** [review]

**Pull request merged**